crates/core/component/shielded-pool/src/fmd.rs

```diff
@@ -59,14 +59,86 @@ impl Default for Parameters {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+/// A struct holding params for the sliding window algorithm
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct SlidingWindow {
+    window_blocks: u32,
+    targeted_detections_per_window: u32,
+}
+
+impl SlidingWindow {
+    pub fn updated_fmd_params(
+        &self,
+        old: &Parameters,
+        state: MetaParametersAlgorithmState,
+        height: u64,
+        clue_count_delta: (u64, u64),
+    ) -> (Parameters, MetaParametersAlgorithmState) {
+        // An edge case, which should act as a constant.
+        if self.window_blocks == 0 {
+            return (
+                old.clone(),
+                MetaParametersAlgorithmState::SlidingWindow {
+                    approximate_clue_count: 0,
+                },
+            );
+        }
+
+        let new_clues_in_block = clue_count_delta.1.saturating_sub(clue_count_delta.1);
```

Comment:
I think this needs to be amended, because this will always be equal to zero

Reply:
also, candidly, i'm working out what index 0 and index 1 correspond to, a short comment would be helpful there

Reply:
maybe destructuring the tuple in the method signature? `(old_clue_count, new_clue_count): (u64, u64)`